exercise-app/src/main/java/org/exercise/app/AppController.java

```diff
@@ -0,0 +1,36 @@
+package org.exercise.app;
+
+import java.io.PrintStream;
+
+import com.examples.attsw.exercise.core.controller.IEmployeeController;
+
+public class AppController {
+
+	private IEmployeeController employeeController;
+
+	public AppController(IEmployeeController employeeController) {
+		this.employeeController = employeeController;
+	}
+
+	public void performAction(String actionCode, String arg, PrintStream out) throws IllegalArgumentException {
+		if (actionCode.equals("showAll") || actionCode.equals("showOne")) {
```

Comment:
Method too long and with too many nested if; refactor in separate methods to make it more readable. Moreover, switch statements could be used (supported for strings since Java 7).

DON'T repeat string literals: make it constants.